packages/next-on-pages/src/buildApplication/buildVercelOutput.ts

```diff
@@ -27,20 +22,24 @@ import {
  *
  */
 export async function buildVercelOutput(): Promise<void> {
-	const pm = await getCurrentPackageManager();
-	cliLog(`Detected Package Manager: ${pm}\n`);
+	const pm = await getPackageManager();
+	if (!pm) {
+		cliError('Error: Could not detect current package manager in use');
+		process.exit(1);
```

Comment:
I generally try to avoid using `process.exit()` deep within code. It makes it difficult to test failure scenarios, as it is hard to mock out and can actually kill the test runner as well as the test subject.
I prefer to use exceptions to bubble up and cause a fatal error towards the top of the application.

Reply:
good points, sure I don't mind changing this, thanks 🙂

Reply:
I've updated this, thanks to your suggestion I noticed that we could actually pass the package manager from the caller (which already needed it anyways), so I moved the check there instead 🙂